### PR TITLE
fix: clippy::size_of_ref

### LINKED ISSUE #1955: Failures in `Linux armv7 uclibceabihf` CI target
The `Linux armv7 uclibceabihf` CI target is failing in multiple unrelated PRs e.g.:
- https://github.com/nix-rust/nix/pull/1948/checks?check_run_id=10378264392
- https://github.com/nix-rust/nix/pull/1945/checks?check_run_id=10378274766
- https://github.com/nix-rust/nix/pull/1947/checks?check_run_id=10378270076

With
```
$TOOL +$TOOLCHAIN clippy $ZFLAGS --target $TARGET --all-targets -- -D warnings
    Checking nix v0.26.1 (/tmp/cirrus-ci-build)
error: argument to `std::mem::size_of_val()` is a reference to a reference
    --> src/sys/socket/mod.rs:1295:17
     |
1295 |                 mem::size_of_val(&iv) + iv.len()
     |                 ^^^^^^^^^^^^^^^^^^^^^
     |
     = help: dereference the argument to `std::mem::size_of_val()` to get the size of the value instead of the size of the reference-type
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#size_of_ref
     = note: `-D clippy::size-of-ref` implied by `-D warnings`

error: argument to `std::mem::size_of_val()` is a reference to a reference
    --> src/sys/socket/mod.rs:1295:17
     |
1295 |                 mem::size_of_val(&iv) + iv.len()
     |                 ^^^^^^^^^^^^^^^^^^^^^
     |
     = help: dereference the argument to `std::mem::size_of_val()` to get the size of the value instead of the size of the reference-type
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#size_of_ref
note: the lint level is defined here
    --> src/lib.rs:46:24
     |
46   | #![cfg_attr(test, deny(warnings))]
     |                        ^^^^^^^^
     = note: `#[deny(clippy::size_of_ref)]` implied by `#[deny(warnings)]`

error: could not compile `nix` due to previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `nix` due to previous error

Exit status: 101
```

### PR DESCRIPTION
**Pull request by JonathanWoollett-Light**
Closes #1955

### CODE REVIEW

**Comment by rtzoeller** [comment]
This change looks correct insofar as it bypasses the clippy warning by being clearer about the intent of the existing code, but I'm not 100% positive the existing code is correct. I'm not very familiar with this part of the nix API though, so I may be mistaken. 

In particular it's not clear to me why we're taking the size of a reference/pointer here, but `af_alg_iv` is larger. Maybe `af_alg_iv.ivlen` is excluded in the length despite being copied?
 
@asomers any ideas?

https://github.com/nix-rust/nix/blob/ed0e859e41fb3611578f759ad2e32a9e98c5dfc0/src/sys/socket/mod.rs#L1210-L1233

**Comment by JonathanWoollett-Light** [comment]
> I'm not 100% positive the existing code is correct. I'm not very familiar with this part of the nix API though, so I may be mistaken.

It also looks a bit off to me, but for now this change should unblock the CI for other PRs. Fixing any issues in the original behaviour can then be a future PR.

File: src/sys/socket/mod.rs [hunk 1]
```diff
@@ -1292,7 +1292,7 @@ impl<'a> ControlMessage<'a> {
             }
             #[cfg(any(target_os = "android", target_os = "linux"))]
             ControlMessage::AlgSetIv(iv) => {
-                mem::size_of_val(&iv) + iv.len()
+                mem::size_of::<&[u8]>() + iv.len()
```
Comment (asomers) [inline src/sys/socket/mod.rs:1295]:
```suggestion
                mem::size_of_val(iv) + iv.len()
```

**Review by asomers — CHANGES_REQUESTED** [review]
It sure seems like the existing code is wrong to me.  I think this would be correct instead.

**Review by rtzoeller — APPROVED** [review]
I'm submitting this to unblock other PRs; I've filed #1960 to track the suspected underlying issue.

bors r+

**Pull request merged**